### DISCUSSION TITLE
Can i change Refiner Layout Vertical to Horizontal?

### DISCUSSION THREAD

**Discussion by Narasireddy447** [microsoft-search/pnp-modern-search#3002] [Ideas]
Is that possible to change the Layout of the Refiner from Vertical to Horizontal in solution? i tried changing using css, but we would like to have mobile/tab again in vertical. we are unable to include @media query in css. 

any inputs, please suggest. please find attached vertical layout. but we need to have Horizontal and in Mobile again in Vertical.
![vertical](https://user-images.githubusercontent.com/63774610/81638926-18994a00-9438-11ea-9053-749424632e18.PNG)

**Comment by wobba** [comment]
Not with the current layout options, which means someone has to contribute a mobile friendly experience.

**Comment by devinprejean** [comment]
Is this referring to the inability to use @media queries in the style tags of template files? Because I can't seem to get an iPhone to recognize anything in that block of CSS inside a style tag.

**Comment by wobba** [comment]
Refiners don't support templating. And mobile has not been focus or tested that much. The filter web part has either vertical or panel+vertical built-in. Horizontal is doable if implemented. Same with more mobile friendly CSS for any web part. Just not been the focus.

**Comment by thastarkmtw** [comment]
Version 4 will support templates & CSS skins in refiners. The version4 branch is a work in progress & lacks the 3.15.1 fixes but you can build it as a preview. Please let me know if you need more information & are willing to test.

**Comment by devinprejean** [comment]
@wobba I'm not trying to change the style of a refiner, just the layout of the results. Is that supported? Like I said I can't get a media query to kick in at all for screen size on the actual iPhone device in either the SP app or Safari

**Comment by wobba** [comment]
@devinprejean if your issue is not related to the original question you should log a separate issue. And then share your CSS and we can troubleshoot.
